File: 2015/01.py
# Advent of Code 2015 - Day 1
# https://adventofcode.com/2015/day/1

# Part 1

with open("inputs/01.txt") as input:
	floor = 0

	for paren in input.read():
		if paren == "(":
			floor += 1
		else:
			floor -= 1

	print(floor)

# Part 2

with open("inputs/01.txt") as input:
	floor = 0

	for i, paren in enumerate(input.read()):
		if paren == "(":
			floor += 1
		else:
			floor -= 1

		if floor == -1:
			print(i + 1) # santa starts at floor 1
			break

File: 2015/01.rb
# Advent of Code 2015 - Day 1
# https://adventofcode.com/2015/day/1

input = File.read 'inputs/01.txt'

# Part 1

floor = 0

input.each_char do |paren|
  floor += if paren == '('
             1
           else
             -1
           end
end

puts floor

# Part 2

floor = 0

input.each_char.with_index(1) do |paren, i|
  floor += if paren == '('
             1
           else
             -1
           end

  break puts(i) if floor == -1
end
